### PR TITLE
Fix the install dir on Windows

### PR DESCRIPTION
**Pull request by leouieda**
Addresses the install path issues pointed out in #62 

Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by seisman** [comment]
@leouieda I don't think this is the right solution.

Comparing the `build.sh` and `bld.bat`, I believe you should add 
```
-D GMT_INSTALL_TRADITIONAL_FOLDERNAMES:BOOL=FALSE
``` 
to windows build script.

**Comment by leouieda** [comment]
Good point @seisman, thanks!

**Comment by leouieda** [comment]
We were missing the `GMT_INSTALL_MODULE_LINKS:BOOL = FALSE` as well. Don't need to create all of the shorthands for GMT4 compatibility.

**Comment by leouieda** [comment]
@seisman what do you think of this?

**Comment by seisman** [comment]
It looks good to me.

---

There is always a warning when running `gmt` on Windows:
```
gmt: Warning: failed to release lock (Error <158>: The segment is already unlocked.)
```
I believe this warning can be fixed by setting `FLOCK` to `on`, but I know nothing about file locking. Maybe @joa-quim can help here.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -41,8 +41,8 @@ requirements:
     - zlib
     - curl
     - pcre
-    - gshhg-gmt
-    - dcw-gmt
+    - gshhg-gmt  # [not win]
+    - dcw-gmt  # [not win]
```
Comment (ocefpaf) [inline recipe/meta.yaml]: :+1: 

Do we know why the Linux/macOS strategy did not work on Windows?

**Comment by joa-quim** [comment]
I don't think there are file locking's on Windows. Again, I've seen that type of warning before. It happened when gmt tried to delete a temporary file but that file was not yet closed. However, I do not see that warning, so once again I don't know how to help.

**Comment by leouieda** [comment]
@seisman is the warning happening to you locally?

File: recipe/meta.yaml (hunk 1 above)
Comment (leouieda) [reply recipe/meta.yaml]: Not really. I have close to zero Windows experience with these things and don't even know where to start looking.

**Comment by seisman** [comment]
> @seisman is the warning happening to you locally?
Yes, it happens locally too.

**Comment by seisman** [comment]
@leouieda As the file lock warning doesn't affect GMT's functioning, we can ignore it at present. I think the PR is good to merge.

**Comment by leouieda** [comment]
:+1: @seisman I just have to bump the build number. After that, we can merge then.

**Pull request merged**